File: sdk/dotnet/src/Cordum.Cap/Bus.cs
using NATS.Client.Core;

namespace Cordum.Cap;

/// <summary>
/// NATS connection configuration.
/// </summary>
public sealed class BusConfig
{
    public string Url { get; set; } = "nats://localhost:4222";
    public string? AuthToken { get; set; }
    public TimeSpan ConnectTimeout { get; set; } = TimeSpan.FromSeconds(5);
}

/// <summary>
/// NATS connection helper.
/// </summary>
public static class Bus
{
    /// <summary>
    /// Connect to NATS using the provided configuration.
    /// </summary>
    public static async Task<NatsConnection> ConnectNats(BusConfig config)
    {
        var opts = new NatsOpts
        {
            Url = config.Url,
            ConnectTimeout = config.ConnectTimeout,
            AuthOpts = config.AuthToken != null
                ? NatsAuthOpts.Default with { Token = config.AuthToken }
                : NatsAuthOpts.Default,
        };

        var connection = new NatsConnection(opts);
        await connection.ConnectAsync();
        return connection;
    }
}

File: sdk/php/src/Bus.php
<?php

declare(strict_types=1);

namespace Cordum\Cap;

use Basis\Nats\Client;
use Basis\Nats\Configuration;

/**
 * NATS connection helper.
 */
final class Bus
{
    /**
     * Connect to NATS using the provided configuration.
     *
     * @param array{
     *   host?: string,
     *   port?: int,
     *   user?: string,
     *   pass?: string,
     *   token?: string,
     *   timeout?: float,
     * } $config Connection configuration
     * @return Client
     */
    public static function connectNats(array $config = []): Client
    {
        $natsConfig = new Configuration([
            'host' => $config['host'] ?? 'localhost',
            'port' => $config['port'] ?? 4222,
            'user' => $config['user'] ?? null,
            'pass' => $config['pass'] ?? null,
            'token' => $config['token'] ?? null,
            'timeout' => $config['timeout'] ?? 5.0,
        ]);

        return new Client($natsConfig);
    }
}
